source/slang/slang-emit-c-like.cpp

```diff
@@ -1150,7 +1153,20 @@ String CLikeSourceEmitter::getName(IRInst* inst)
     String name;
     if(!m_mapInstToName.tryGetValue(inst, name))
     {
-        name = generateName(inst);
+        // unmangle names, when emitting header
+        if (shouldEmitOnlyHeader())
+        {
+            if (auto nameHintDecor = inst->findDecoration<IRNameHintDecoration>())
```

Comment:
This feels hacky. We should use the same name emit logic for header and cpp files. For things that needs to be emitted without name mangling, they should be marked as extern_cpp.